agent/utils/ephemeral_ports.go

```diff
@@ -114,10 +128,15 @@ func GetHostPort(protocol string, dynamicHostPortRange string) (string, error) {
 	defer portLock.Unlock()
 	numberOfPorts := 1
 	result, err := getNumOfHostPorts(numberOfPorts, protocol, dynamicHostPortRange)
+	foundHostPort := strings.Split(result, "-")[0]
 	if err == nil {
-		result = strings.Split(result, "-")[0]
+		// Verify the found host port is within the given dynamic host port range
```

Comment:
__NEW!__: Add port validation as ports should be validated before returning back from  `utils.GetHostPort()`